### ISSUE TITLE
Prioritization of different typographic features

### ISSUE THREAD

**Issue by xfq** [w3c/clreq#458]
We can consider marking the importance of different typographic features in some way (kind of like the WCAG conformance levels), such as:

* Must-have features
* Important features
* Features only used in advanced publishing

I think this information is useful to spec writers, {browser, e-book reader, word processor, desktop publishing software} vendors, and content authors/developers alike, because they can refer to the importance levels to determine priorities when writing specs, implementing specs, and creating content using features in software.

Currently, some prioritization information can be seen in the [Gap Analysis document](https://www.w3.org/International/clreq/gap-analysis/), but it is gap-oriented and not instructive enough (for example, for a vendor trying to implement a new browser).

Here are a few example features:

### Must-have features

* [Support rendering of all Simplified Chinese and Traditional Chinese punctuations](https://w3c.github.io/clreq/#categories_and_usage_of_punctuation_marks)

### Important features

* [Increased inter-character spacing](https://w3c.github.io/clreq/#increased_inter_character_spacing)
* [Support prohibition rules for line start and line end](https://w3c.github.io/clreq/#prohibition_rules_for_line_start_end)
* [Support prohibition rules for unbreakable marks](https://w3c.github.io/clreq/#prohibition_rules_for_unbreakable_marks)
* [Extra spacing between adjacent Han characters and Western characters](https://w3c.github.io/clreq/#mixed_text_composition_in_horizontal_writing_mode)
* [Support mixed text composition in vertical writing mode](https://w3c.github.io/clreq/#mixed_text_composition_in_vertical_writing_mode)
* [Support Bopomofo](https://w3c.github.io/clreq/#positioning_of_zhuyin)
* [Support pinyin](https://w3c.github.io/clreq/#positioning_of_romanization)
* [Paragraph adjustment](https://w3c.github.io/clreq/#paragraph_adjustment_rules)
* [Line adjustment](https://w3c.github.io/clreq/#line_adjustment)

### Features only used in advanced publishing

* [Reduced inter-character spacing](https://w3c.github.io/clreq/#reduced_inter_character_spacing)
* [Punctuation width adjustment](https://w3c.github.io/clreq/#punctuation_width_adjustment)
* [Support interlinear comments](https://w3c.github.io/clreq/#positioning_of_interlinear_comments)

**Comment by xfq** [comment]
Things that can affect the importance of a typographic feature include whether it is common, whether it affects legibility, whether it violates national standards or other standards, and so on.

**Comment by kecrily** [comment]
Prioritization is really important. Everyone can know "what is the most important thing I need to do now".

---

Planning on rearranging existing content or making a separate priority list?

**Comment by xfq** [comment]
I prefer maintaining this separately from the main document.

That said, if anyone comes up with a good presentation/structure, I am not against writing this information directly in the main clreq document.

**Comment by r12a** [comment]
It could be an Appendix. Just an idea. It seems useful to keep it in the same document since it is more readily findable that way, and more likely to be updated as the main document changes.

**Comment by macnmm** [comment]
This is a common ask for me so I thought I would weigh in :)

Japanese layout (and I suspect Korean and Chinese as well) even in the simplest, most unsophisticated case, has many possible variations, or "right answers" for how it should be set. This often varies with the design intent, the actual content, and other aspects that require there to be many options be made available in the layout engine beyond those normally exposed for simple Latin text use cases.

For example, in more traditional Japanese typography, the box width should be set to an integer multiple of the point size (type body/embox width) and the line fully-justified so the edges match, and the lines with only full-width characters end up being set flush because they already fit perfectly. However, this is not the one-size-fits-all default for Japanese text; modern electronic store ad copy, for example, is often set proportionally with very tight leading and different font sizes with different alignments and even uneven font scaling or other distortion. This implies the need for things like x/y scaling, baseline shifting, alignment to the embox edges, ruby annotations, and other things not usually required of Latin script layout until you get to some rare case of fanciness not needed in an email message for example. But, if you do not support these basics for Japanese, you are not really supporting the MVP, in my opinion. Japanese being a script that draws its history from the first movable type of square characters that then had to be spaced and aligned and distorted differently to achieve harmony with what is being written and for what purpose, has a higher level of requirements than Latin text, and in expressing what is the list of prioritized features of Japanese layout I feel we must remember there is a wide spectrum of variation in Japanese text layout for any purpose, even the most "simple". I wonder if this is also true for Chinese layout use cases?
